test/built-ins/TypedArray/prototype/sort/sorted-values-nan.js

```diff
@@ -0,0 +1,31 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-%typedarray%.prototype.sort
+description: Sort values to numeric ascending order
+info: >
+  22.2.3.26 %TypedArray%.prototype.sort ( comparefn )
+
+  When the TypedArray SortCompare abstract operation is called with two
+  arguments x and y, the following steps are taken:
+
+  ...
+
+  NOTE: Because NaN always compares greater than any other value, NaN property
+  values always sort to the end of the result when comparefn is not provided.
+includes: [testTypedArray.js, compareArray.js]
+---*/
+
+testWithTypedArrayConstructors(function(TA) {
+  var sample;
+
+  sample = new TA([2, NaN, NaN, 0, 1]).sort();
+  assert(compareArray(sample.slice(0, 3), [0, 1, 2]), "case #1");
```

Comment:
Maybe I'm being too sensitive, but I'm concerned about using different
  prototype methods to test this one because consumers may not have those
  available yet. Would you mind writing out the assertions in "long form"?